### DISCUSSION TITLE
Preceding numbers are skipped.

### DISCUSSION THREAD

**Discussion by YazeedAlsaif** [koenhendriks/laravel-str-acronym#5] [Ideas]
Hi 

Thank you guys for your work :)

I have been using your package and found an inconsistency in acronym generation for company names. For instance, when I input "21st Century Fox," the generated acronym is "sCF," whereas the correct acronym for the company is "21CF." Similarly, when I input "Forever 21," the generated acronym is "F," whereas the correct acronym for the company is "F21."

I am not sure, grammatically, if we should skip preceding or trailing digits in generating acronyms from company names, as it may affect the grammatical correctness of the acronym. I wanted to bring this issue to your attention so that it can be addressed, if you think it should.

Thank you again.

**Comment by koenhendriks** [comment]
This is an interesting case. 

When I build this package I figured we don't want any numbers in the acronyms, as is confirmed by [this line in the tests](https://github.com/koenhendriks/laravel-str-acronym/blob/main/tests/AcronymStrTest.php#L22). 

Acronyms typically do not contain numbers. Acronyms are formed by using the initial letters or parts of a series of words to create a pronounceable word. While it is possible to include numbers in an acronym, it is less common. Numbers are usually used in initialisms, which are similar to acronyms but are pronounced as individual letters. For example, "NASA" (National Aeronautics and Space Administration) is an acronym, while "IBM" (International Business Machines) is an initialism. '

Implementing these would in my opinion goes against the scope of the package. But I'm open for the discussion :)

**Comment by cornedor** [comment]
If this is being implemented, I suppose this should be configurable. There are indeed use cases, like the 21CF, where it is much better to include numbers. But for example, in a use case where large numbers could be common, you don't want this. Say, you want to generate acronyms for product names, and product names could contain serial numbers, those acronyms could get very long.

**Comment by koenhendriks** [comment]
So in my initial use case I was generating 'icons' in a list of events with the acronym of the given event title. I made a quick tailwind snippit to illustrate what I mean:

https://play.tailwindcss.com/E5D7hPgjFU

For me it looked off to have the numbers in the 'icons' so I build the package to exclude those.

![image](https://github.com/koenhendriks/laravel-str-acronym/assets/4124579/2b553028-d494-40d4-906b-f07261378f5f)
In this example we see the list as I preferred in my use case. 

Enabling numbers would create this:
![image](https://github.com/koenhendriks/laravel-str-acronym/assets/4124579/e0762957-6b7f-49cd-b98e-3e862ce8808b)
Which I wouldn't want.

I guess there could be cases for with and without numbers, but then it creates the question to have this as a global config? Or should we create another macro for the `Str` class; i.e. `Str::numberedAcronym('text')`?

**Reply by cornedor** [reply]
Making it a global config will limit your whole project to only one type. Perhaps deprecate the delimiter parameter in favor of an object, so you can pass multiple options; e.g.

```
Str::acronym('21th Century Fox', [
  'delimiter' => '-',
  'includeNumbers' => true
]);

```
